backend/src/packages/meditation/meditation.repository.ts

```diff
@@ -0,0 +1,78 @@
+import { type Repository } from '#libs/types/types.js';
+
+import { MeditationRelation } from './libs/enums/enums.js';
+import {
+  type MeditationCommonQueryResponse,
+  type MeditationCreateQueryPayload,
+  type TopicCommonQueryResponse,
+} from './libs/types/types.js';
+import { MeditationEntity } from './meditation.entity.js';
+import { type MeditationEntriesModel } from './meditation-entries.model.js';
+import { type MeditationTopicModel } from './meditation-topics.model.js';
+
+class MeditationRepository implements Repository {
+  private meditationEntryModel: typeof MeditationEntriesModel;
+  private meditationTopicModel: typeof MeditationTopicModel;
+
+  public constructor(
+    meditationEntryModel: typeof MeditationEntriesModel,
+    meditationTopicModel: typeof MeditationTopicModel,
+  ) {
+    this.meditationEntryModel = meditationEntryModel;
+    this.meditationTopicModel = meditationTopicModel;
+  }
+
+  public find(): ReturnType<Repository['find']> {
+    return Promise.resolve(null);
+  }
+
+  public async findTopicByName(
+    name: string,
+  ): Promise<TopicCommonQueryResponse | undefined> {
+    return await this.meditationTopicModel
+      .query()
+      .findOne({
+        name,
+      })
+      .castTo<TopicCommonQueryResponse | undefined>();
+  }
```

Comment:
```suggestion
```

Reply:
@Liza-Veis Should I leave meditation topic model or I can delete it for now?